### PR TITLE
Fix doc cosineannealinglr 152081

### PR DESCRIPTION
**Pull request by Jacobgoss30**
## Summary

This PR updates the docstring for `CosineAnnealingLR` to accurately reflect its recursive learning rate schedule. The previous docstring displayed only the SGDR closed-form expression, which doesn't match the actual recursive implementation in code.

Changes:

- Added the recursive update formula used in `get_lr()`
- Retained the original closed-form SGDR expression for reference
- Clarified that warm restarts are not implemented in this scheduler

This addresses confusion raised in issue #152081.

## Related issue

[#152081](https://github.com/pytorch/pytorch/issues/152081)

## Testing

Doc-only change. Ran pre-commit to verify formatting.

### CODE REVIEW

File: torch/optim/lr_scheduler.py [hunk 1]
```diff
@@ -1000,51 +1000,54 @@ def _get_closed_form_lr(self):
...
+    - :math:`T_{cur}` is the number of epochs since the last restart
+    - :math:`T_{max}` is the maximum number of epochs in a cycle
+
+    Note:
+        Although SGDR includes periodic restarts, this implementation performs cosine annealing
+        **without restarts**, so :math:`T_{cur} = t` and increases monotonically with each call
+        to :meth:`step`.
+
+    To retrieve the exact closed-form learning rate at any point, use the
+    :meth:`_get_closed_form_lr` method.
```
Comment (janeyx99) [inline torch/optim/lr_scheduler.py:1032]: Let's not advertise private functions as we'd like to move people off of it

**Review by janeyx99 — COMMENTED** [review]
Looks good so far--thank you for the clear explanations! In both the code and the issue comment.

Could you paste a screenshot of before and after for the docs? (I've approved the preview so you should be able to get to the python doc preview when it becomes ready ~40m from the pytorchbot comment on this PR.)

**Comment by Jacobgoss30** [comment]
@janeyx99 I have updated the documentation based on your feedback. Please can you trigger the CI actions to rebuild the preview so I can share the final before and after

**Comment by Jacobgoss30** [comment]
New:
Screenshot 2025-05-08 at 10 32 02
Screenshot 2025-05-08 at 10 32 13

Previous:
Screenshot 2025-05-08 at 10 33 07

**Review by janeyx99 — COMMENTED** [review]
looks good

**Review by janeyx99 — APPROVED** [review]

**Comment by bbbbbbbbba** [comment]
Shouldn't we mention that `T_{cur} = (2k+1)T_{max}` is still a special case? (Otherwise it would be division by zero.)
https://github.com/pytorch/pytorch/blob/1eba9b3aa3c43f86f4a2c807ac8e12c4a7767340/torch/optim/lr_scheduler.py#L1117-L1122
